src/LogicTasks/Syntax/TreeToFormula.hs

```diff
@@ -76,9 +77,17 @@ partialGrade _ sol
 
 completeGrade :: OutputMonad m => SynTreeInst -> TreeFormulaAnswer -> LangM m
 completeGrade inst sol
-    | fromJust ( maybeTree sol) /= tree inst = reject $ do
-      english "Your solution is not correct."
-      german "Ihre Abgabe ist nicht die korrekte Lösung."
+    | fromJust ( maybeTree sol) /= tree inst = refuse $ do
+      instruct $ do
+        english "Your solution is incorrect."
+        german "Ihre Lösung ist falsch."
+
```

Comment:
My intention in https://github.com/fmidue/logic-tasks/issues/35#issue-1579514064 for this task was (but not appropriately expressed there originally, due to my Boolean blindness) that if:
- the student gave an incorrect answer,
- and independently of whether `showSolution` was set,

the syntax tree that would correspond to *their* (wrong) solution formula would be printed.

(And additionally, but only if their answer was wrong and `showSolution = True`, the correct formula is printed, as already implemented below.)

Reply:
I merged the current master into this branch and updated `completeGrade`. This should now work as you desired.